File: implementations/javascript/matrix_multiplication.js
const SIZES = { small: 50, medium: 200, large: 500 };
const n = SIZES[process.argv[2]];

const a = Array.from({ length: n }, (_, i) => Array.from({ length: n }, (_, j) => i + j));
const b = Array.from({ length: n }, (_, i) => Array.from({ length: n }, (_, j) => i - j));
const c = Array.from({ length: n }, () => new Array(n).fill(0));

for (let i = 0; i < n; i++) {
    for (let j = 0; j < n; j++) {
        let acc = 0;
        for (let k = 0; k < n; k++) acc += a[i][k] * b[k][j];
        c[i][j] = acc;
    }
}

console.log(c[Math.floor(n / 2)][Math.floor(n / 2)]);

File: implementations/python/matrix_multiplication.py
import sys

SMALL  = 50
MEDIUM = 200
LARGE  = 500

SIZES = {"small": SMALL, "medium": MEDIUM, "large": LARGE}

if __name__ == "__main__":
    n = SIZES[sys.argv[1]]
    a = [[i + j for j in range(n)] for i in range(n)]
    b = [[i - j for j in range(n)] for i in range(n)]
    c = [[0] * n for _ in range(n)]
    for i in range(n):
        for j in range(n):
            acc = 0
            for k in range(n):
                acc += a[i][k] * b[k][j]
            c[i][j] = acc
    print(c[n // 2][n // 2])
